src/nvim/ops.c

```diff
@@ -3644,6 +3644,10 @@ int do_join(size_t count,
    */
   for (t = (linenr_T)count - 1;; t--) {
     cend -= currsize;
+	if(cend == NULL || curr == NULL){
```

Comment:
`cend` was dereferenced on line 3636, so it can't be NULL here.

AFAICT `curr` cannot be NULL unless count=0. And the docstring for the function claims that `count` minimum is 2. 

So I would the only thing needed is 

    assert(cound >= 2); 

near the start of the function.